### PR TITLE
[opencv4] update to 4.9.0

### LINKED ISSUE #35937: [opencv4] update to 4.9.0
### Library name

opencv4

### New version number

4.9.0

### Other information that may be useful (release notes, etc...)

https://github.com/opencv/opencv/releases/tag/4.9.0

_No response_

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fixes #35937

Update ports opencv4 to the latest version 4.9.0

- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by russelltg** [comment]
It probably makes sense to enable OpenCV 4.9's new native CMake CUDA support when building against cuda, as it's much easier to configure with cmake toolchains. `-DENABLE_CUDA_FIRST_CLASS_LANGUAGE=ON`

**Comment by JonLiu1993** [comment]
> It probably makes sense to enable OpenCV 4.9's new native CMake CUDA support when building against cuda, as it's much easier to configure with cmake toolchains. `-DENABLE_CUDA_FIRST_CLASS_LANGUAGE=ON`. It should be safe to just unconditionally set it, even when compiling without cuda.

Thanks for the reminder, I will update the PR

File: ports/opencv4/portfile.cmake [hunk 1]
```diff
@@ -1,17 +1,10 @@
 set(USE_QT_VERSION "6")
 
-# https://github.com/opencv/opencv/pull/24043
```
Comment (JonLiu1993) [inline ports/opencv4/portfile.cmake:3]: This PR has been applied to the latest version

File: ports/opencv4/portfile.cmake [hunk 2]
```diff
@@ -23,14 +16,9 @@ vcpkg_from_github(
...
       0009-fix-protobuf.patch
       0010-fix-uwp-tiff-imgcodecs.patch
-      0011-remove-python2.patch
       0012-fix-zlib.patch
       0015-fix-freetype.patch
       0017-fix-flatbuffers.patch
-      0019-missing-include.patch
-      0020-fix-compat-cuda12.2.patch
-      0021-static-openvino.patch # https://github.com/opencv/opencv/pull/23963
-      "${ARM64_WINDOWS_FIX}"
```
Comment (JonLiu1993) [inline ports/opencv4/portfile.cmake:33]: This PR has been applied to the latest version

File: ports/opencv4/portfile.cmake [hunk 3]
```diff
@@ -512,6 +500,7 @@ find_dependency(Threads)")
   endif()
   if("cuda" IN_LIST FEATURES)
     string(APPEND DEPS_STRING "\nfind_dependency(CUDA)")
+    list(APPEND ADDITIONAL_BUILD_FLAGS "-DENABLE_CUDA_FIRST_CLASS_LANGUAGE=ON")
```
Comment (russelltg) [inline ports/opencv4/portfile.cmake:503]: I think this needs to come before the configure

**Comment by clibdev** [comment]
Is someone still working on this PR? I'm considering creating a new PR for OpenCV 4.9.

**Comment by JonLiu1993** [comment]
> Is someone still working on this PR? I'm considering creating a new PR for OpenCV 4.9.

You can resubmit a new PR, thank you.

**Comment by hesmar** [comment]
@clibdev Are you working on a new PR?

**Comment by JonLiu1993** [comment]
Duplicate of #38928.

**Pull request closed without merging**